src/config/loading.rs

```diff
@@ -296,6 +215,72 @@ fn load_builder_from_dir(
             }
         }
     }
+
+    let subfolder = path.join("enrichment_tables");
+    if subfolder.exists() && subfolder.is_dir() {
+        let (value, warns) = super::recursive::load_dir(&subfolder)?;
```

Comment:
do we really want/need to support recursive loading for anything but transforms?

Reply:
I asked myself the same question and ended choosing to have the same behavior everywhere. But if @binarylogic or @jszwedko think it's not needed, I'm ok to drop that 😉

Reply:
Agree, enrichment table names can't be nested, right? So I see no reason to allow recursion.

Reply:
Yeah, I'm fine with dropping that until we find a need.

Reply:
What about the others? If I follow the logic, I'll drop it for the others as well.

Reply:
I fixed it in [here](https://github.com/vectordotdev/vector/pull/10179/commits/8e6cdd75ab63b5a192463d5b822ad568e80d045c) by not using the `recursive` method, so that we don't loose the validation from `serde`.